standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -353,11 +353,9 @@ public void markCompacted(CompactionInfo info) throws MetaException {
         if (minOpenTxnWaterMark > 0) {
           whereClause += " AND (\"CQ_NEXT_TXN_ID\" <= " + minOpenTxnWaterMark + " OR \"CQ_NEXT_TXN_ID\" IS NULL)";
         }
-        if (retentionTime > 0) {
-          whereClause += " AND \"CQ_COMMIT_TIME\" < (" + getEpochFn(dbProduct) + " - " + retentionTime + ")";
-        }
+        whereClause += " AND (\"CQ_COMMIT_TIME\" < (" + getEpochFn(dbProduct) + " - CQ_RETRY_RETENTION - " + retentionTime + ") OR \"CQ_COMMIT_TIME\" IS NULL)";
```

Comment:
It would probably be best to fix the `if (retentionTime > 0)` removal in a separate ticket since it fixes an unrelated bug

Reply:
This is my plan, but unfortunately this change cannot be separated. From now on the CQ_COMMIT_TIME comparisaon needs to be done all the case (because of CQ_RETRY_RETENTION), regardless of the value of the retention time. If I would keep the if condition, retry retention would not work if the cleaner retention time is 0. What I plan to do is to fix the where clause in a separate ticket too, and this PR should be merged to master only after that one, mimicking that the fix (adding the _OR "CQ_COMMIT_TIME" IS NULL_ part) existed before, and this PR built on top of that fix. What do you think?

Reply:
Yes, it would be best to fix the where clause in a separate ticket, commit that first, and rebase this change on top of that.